### PR TITLE
commands/pin: use new cmds lib

### PR DESCRIPTION
**Pull request by overbool**
Refer: https://github.com/ipfs/go-ipfs/issues/5664

License: MIT
Signed-off-by: Overbool <overbool.xu@gmail.com>

### CODE REVIEW

File: core/commands/pin.go [hunk 1]
```diff
@@ -65,118 +64,109 @@ var addPinCmd = &cmds.Command{
...
-				if res.Error() != nil {
-					return nil, res.Error()
-				}
-			default:
-				return nil, e.TypeErr(out, v)
+			if out.Pins != nil {
+				added = out.Pins
+			} else {
+				// this can only happen if the progress option is set
+				return fmt.Errorf("Fetched/Processed %d nodes\r", out.Progress)
```
Comment (kevina) [inline core/commands/pin.go]: This is wrong.  This is not an error.

This command should likely use a PostRun.

**Comment by overbool** [comment]
@kevina I found that req.Arguments only contain the first arg if I pass two arguments from Stdin

**Comment by kevina** [comment]
> @kevina I found that `req.Arguments` only contain the first arg if I pass two arguments from Stdin.

Correct.  To fix this you need to call `req.ParseBodyArgs()` before using `req.Arguments`.

**Comment by kevina** [comment]
@overbool this needs a rebase now.

I would like to get this is ASAP.  Let me know if you need any help with this.

**Comment by overbool** [comment]
@kevina sorry for my delay. I had replaced `Encoder` with `PostRun` and fix some bugs. But, This `pr` seems to have a strange problem that it can't sync my latest commit.

**Comment by overbool** [comment]
@kevina sorry for my delay. I had replaced the `Encoder` with `PostRun` and fixed some bugs. Could u help me review it again?

**Comment by Stebalien** [comment]
So, technically, this breaks `curl http://localhost:5001/api/v0/pin/add/...?enc=text`. I believe the correct way to do this is: https://github.com/overbool/go-ipfs/pull/1. That is, handle status in a post-run but forward everything else to the encoder.

**Pull request merged**